### DISCUSSION TITLE
Ability to set custom frequencies based on which governor is set

### DISCUSSION THREAD

**Discussion by JoshuaKimsey** [JakeRoggenbuck/auto-clock-speed#543] [General]
So I've been using auto-clock-speed for about a day now and do quite like the features it brings for monitoring what my CPU is doing. This is especially helpful since I moved away from Gnome where I used an extension to handle this for me.

One question I had though, is there a way to set custom frequencies for when different governors are active? I ask, because with regular auto-cpufreq, it tried to do this, but the limits were simply ignored on my fresh install of OpenSuse Tumbleweed. And since I chose KDE over Gnome, I no longer have access to the extension I used previously to set custom frequencies for my machine (ThinkPad X1 Yoga Gen 6 - 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz). It would be nice to be able to set a custom frequency for battery vs charging, as that would help to really preserve my battery life. Currently with just auto-clock-speed installed (and TLP uninstalled from where I previously worked with auto-cpufreq), it's just drinking the battery still. Easily a percentage point every 2-3 minutes doing nothing. What I would love to do is, on battery power, turn off turbo-boosting, and probably limit the core frequencies down some, probably to 2000MHz, to really help extend battery life. As it is now, the max is 4800 with turbo set to on, which doesn't save any battery life for me.

If there is a way to reign this in further with some custom frequencies, let me know! I didn't see a way to do this in the documentation, and I didn't see anything in the acs.toml file for managing this either? Not even sure what states are available to set honestly. Love what this project is trying to do though, and especially love that it's being done in Rust! :)

**Comment by JakeRoggenbuck** [comment]
Great question and great feedback! So far it just adjusts to powersave or performance based on load, usage, and battery capacity. This changes the frequency max and min some amount automatically but I definitely agree it would be awesome to have more granular control in the config. Thanks for that amazing idea! I will make an issue and work on this over the next week.

**Reply by JakeRoggenbuck** [reply]
Created the issue #544
